backend/src/main/java/utils/WordPhoneticGetter.java

```diff
@@ -0,0 +1,79 @@
+package com.animalrhymes.server;
+
+import com.jayway.jsonpath.*;
+import java.util.*;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/*
+ * Class with one sole purpose: return the phonetic of some word using utility method getPhonetic(String word)
+ */
+public class WordPhoneticGetter {
+    private static final String USER_AGENT = "Mozilla/5.0";
+    private static final String GET_URL = "https://od-api.oxforddictionaries.com:443/api/v2/entries/en-us/";
+	private static final String PRONUNCIATION_FIELD = "fields=pronunciations";
+	private static int current = 0;
+
+    public static String getPhonetic(String word) {
+    	// API only handles 500 rpm. This is a lazy way to handle this.
+    	current += 1;
+    	if (current == 499) {
+			current = 0;
```

Comment:
Seems like you might still go over the rate limit if you set current to 0 before sleeping. Request 1 could come in from the frontend when current is 499, set current to 0, and then either return a delayed response or timeout. Request 2 could come in right after Request 1, read current as 0, and proceed, hitting the limit. 

It's not something I'd worry too much about for an MVP

Reply:
You're so right